### PR TITLE
feat: clean navigation optionMatch function

### PR DESCRIPTION
**Pull request by loc-nguyenthien**
Remove the argument of `RouteKey.<ScreenName>` we have to pass into `optionMatch` function

### CODE REVIEW

File: template/src/navigation/ScreenService.js [hunk 1]
```diff
@@ -17,15 +17,36 @@ export const screenMatch = screen => {
...
-      return {
-        headerLeft: null,
-      }
+      headerLeft = null
+      break
     default:
-      return {}
+      break
+  }
+  // Screen Options
```
Comment (honghoangsts) [inline template/src/navigation/ScreenService.js:39]: I don't think we need generate screen options any more

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: template/src/navigation/ScreenService.js (hunk 1 above)
Comment (loc-nguyenthien) [reply template/src/navigation/ScreenService.js:39]: @honghoangsts I think we still need this, in order to configure for `options` for screens
This PR is just to remove the need to pass `RouteKey.<ScreenName>` inside this option generator function

Comment (honghoangsts) [reply template/src/navigation/ScreenService.js:39]: I don't think all screen need to add to this function. There are some specific case need to change the header options.

Comment (loc-nguyenthien) [reply template/src/navigation/ScreenService.js:39]: Well this is just to put all the custom options for screens in one place. If those screen that don't need to change option can ignore this. But I think we still need it here for other screen that need changing

Comment (loc-nguyenthien) [reply template/src/navigation/ScreenService.js:39]: So maybe I will remove the default return object and let every switch case return their own value. Like this
```
export const optionsMatch = ({route}) => {
  const name = route?.name
  switch (name) {
    case RouteKey.HomeScreen:
    case RouteKey.HomeStack:
      return {
        headerLeft: null,
      }
      break
    default:
      return {}
  }
}
```

Comment (loc-nguyenthien) [reply template/src/navigation/ScreenService.js]: @honghoangsts updated auto generate screen scripts

**Review by vunguyendhSTS — APPROVED** [review]

**Review by ngochuyduong — APPROVED** [review]

**Review by anhle10051996 — APPROVED** [review]

**Review by TranQuangPhi — APPROVED** [review]

**Review by honghoangsts — APPROVED** [review]

**Pull request merged**